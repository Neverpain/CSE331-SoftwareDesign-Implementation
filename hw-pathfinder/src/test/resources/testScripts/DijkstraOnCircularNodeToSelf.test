# Tests finding the path using Dijkstra's algorithm from a node, to another node
# than back to the original node and prints out the path

CreateGraph g

AddNode g A
AddNode g B
AddNode g C

AddEdge g A B 1.0
AddEdge g B C 1.0
AddEdge g C A 1.0

FindPath g A A

FindPath g A B

FindPath g A C

FindPath g C A