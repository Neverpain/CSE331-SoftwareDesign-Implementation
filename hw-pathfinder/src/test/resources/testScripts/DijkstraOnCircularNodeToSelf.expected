# Tests finding the path using Dijkstra's algorithm from a node, to another node
# than back to the original node and prints out the path

created graph g

added node A to g
added node B to g
added node C to g

added edge 1.000 from A to B in g
added edge 1.000 from B to C in g
added edge 1.000 from C to A in g

path from A to A:
total cost: 0.000

path from A to B:
A to B with weight 1.000
total cost: 1.000

path from A to C:
A to B with weight 1.000
B to C with weight 1.000
total cost: 2.000

path from C to A:
C to A with weight 1.000
total cost: 1.000